A5.md

```diff
@@ -0,0 +1,157 @@
+Load Balancing and DNS
+----
+* Author(s): Mark D. Roth (roth@google.com)
+* Approver: a11r
+* Status: Draft
+* Implemented in: <language, ...>
+* Last updated: 2017-01-31
+* Discussion at: https://groups.google.com/d/topic/grpc-io/6be1QsHyZkk/discussion
+
+## Abstract
+
+This document describes how additional name service data for load
+balancing should be encoded in DNS.
+
+## Background
+
+As described in the [name
+resolution](https://github.com/grpc/grpc/blob/master/doc/naming.md) doc,
+a resolver implementation is expected to return two additional pieces of
+data with each address, both related to [load
+balancing](https://github.com/grpc/grpc/blob/master/doc/load-balancing.md).
+Those two pieces of data are:
+
+- A boolean indicating whether the address is a backend address (i.e.,
+  the address to use to contact the server directly) or a balancer
+  address.
+
+- The name of the balancer, if the address is a balancer address.
+  This will be used to perform peer authorization.
+
+This document describes how those two pieces of data should be encoded
+when the name service is DNS.
+
+### Related Proposals: 
+
+N/A
+
+## Proposal
+
+This proposal makes use of DNS SRV records, as described in
+[RFC-2782](https://tools.ietf.org/html/rfc2782).  The SRV record for
+grpclb will use the following values:
+
+- service name: `grpclb`
+- protocol: `tcp`
+- priority: this will be `0` for all grpclb SRV records
+- weight: this will be `0` for all grpclb SRV records
+
+When the gRPC client library opens a channel for a given server name,
+it will ask DNS for these SRV records in addition to the normal address
+records.  For each SRV record that comes back in the response, the
+client library will then do a DNS lookup for address records for the name
+specified by the SRV record.  It will then return each of the resulting
+addresses with the extra fields indicating that the addresses are
+balancer addresses and the name from the SRV record.
+
+### Example
+
+For example, let's say that we have the following DNS zone file:
+
+```
+$ORIGIN example.com.
+; grpclb for server.example.com goes to lb.example.com:1234
+_grpclb._tcp.server IN SRV 0 0 1234 lb
+; lb.example.com has 3 IP addresses
+lb                  IN A 10.0.0.1
+                    IN A 10.0.0.2
+                    IN A 10.0.0.3
+```
+
+With this data, a gRPC client will resolve the name `server.example.com`
+to the following addresses:
```

Comment:
Unless I'm misunderstanding something from the SRV RFC, I think that records with the same priority and weight are all equally likely to be chosen, which is exactly the behavior that we're supporting here.  So I don't think there's any surprise here.

That having been said, I have added a paragraph explaining that we do not support any priority or weighting.